### PR TITLE
Publish make-host-app-editable command with CLI docs.

### PR DESCRIPTION
**Pull request by matthew-carroll**
DON'T MERGE: Until @gspencergoog's work on `flutter create` (#22565) is merged.

Publish make-host-app-editable command with CLI docs.

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/make_host_app_editable.dart [hunk 1]
```diff
@@ -10,69 +10,57 @@ import '../runner/flutter_command.dart';
...
-  final String name = 'make-host-app-editable';
+  FlutterProject _project;
 
   @override
-  final String description = 'Commands for making host apps editable within a Flutter project';
+  final String name = 'make-host-app-editable';
 
   @override
-  bool get hidden => true;
+  final String description = 'Converts generated host apps to editable, non-generated host apps.\n\n'
```
Comment (dnfield) [inline packages/flutter_tools/lib/src/commands/make_host_app_editable.dart]: This line is a little confusing at first read - maybe something like `'Moves the Flutter controlled and generated host app(s) to a user controlled location.\n\n'`?

File: packages/flutter_tools/lib/src/commands/make_host_app_editable.dart [hunk 2]
```diff
@@ -10,69 +10,57 @@ import '../runner/flutter_command.dart';
...
-
-  @override
-  String get description => 'Make an iOS host app editable within a Flutter project';
+    await _project.ensureReadyForPlatformSpecificTooling();
 
-  @override
-  Future<Null> runCommand() async {
-    await super.runCommand();
-    await _project.ios.makeHostAppEditable();
+    final bool isAndroidRequested = argResults.arguments.contains('android');
```
Comment (dnfield) [inline packages/flutter_tools/lib/src/commands/make_host_app_editable.dart]: why not just `argResults['android']`?

File: packages/flutter_tools/lib/src/commands/make_host_app_editable.dart [hunk 3]
```diff
@@ -10,69 +10,57 @@ import '../runner/flutter_command.dart';
...
+    final bool isIOSRequested = argResults.arguments.contains('ios');
+
+    if (isAndroidRequested) {
```
Comment (dnfield) [inline packages/flutter_tools/lib/src/commands/make_host_app_editable.dart]: This logic will do nothing if a user decides to pass both `--ios` and `--android`, right?  Seems like we might need to check for the case where both are specified explicitly.

File: packages/flutter_tools/lib/src/commands/make_host_app_editable.dart (hunk 1 above)
Comment (matthew-carroll) [reply packages/flutter_tools/lib/src/commands/make_host_app_editable.dart]: Tried some different wording

File: packages/flutter_tools/lib/src/commands/make_host_app_editable.dart (hunk 2 above)
Comment (matthew-carroll) [reply packages/flutter_tools/lib/src/commands/make_host_app_editable.dart]: Done

File: packages/flutter_tools/lib/src/commands/make_host_app_editable.dart (hunk 3 above)
Comment (matthew-carroll) [reply packages/flutter_tools/lib/src/commands/make_host_app_editable.dart]: It would do something, but not the right thing.  I've updated the logic.

Comment (dnfield) [reply packages/flutter_tools/lib/src/commands/make_host_app_editable.dart]: Should `if (isAndroidRequested == isIOSRequested) {` be `if (isAndroidRequested == false && isIOSRequested == false) {`?  Or does it not hurt anything to run these commands multiple times (they immediately exit if already done)?

**Review by dnfield — APPROVED** [review]
LGTM

**Review by gspencergoog — APPROVED** [review]

**Comment by mit-mit** [comment]
Are we ready to merge this?

**Comment by matthew-carroll** [comment]
@mit-mit Just merged.  Thanks.

**Pull request merged**